### ISSUE TITLE
High contrast colours stylesheet tweaks

### ISSUE THREAD

**Issue by Restioson** [Restioson/vertex#43]
It would be great if someone else more knowledgeable about accessible colour design could look over our styles. Changes that are visually appealing regardless of a11y concerns can be implemented directly into the main stylesheet, whereas ones that are not so appealing can be added to the high contrast colours stylesheet.

**Comment by natastro** [comment]
Hi @Restioson ! I'd love to look into this one for you. Thank you!

**Comment by Restioson** [comment]
Awesome, thanks for the offer! Do you need any help getting the client up and running? What platform are you on? I think I will need to update the public testing server to the latest dev branch.

**Comment by natastro** [comment]
Hi @Restioson ! If you are able to help me get the client up and running, that would be great! Or - if it's easier, I can assist if you have screenshots. Whatever works for you! Take care.

**Comment by Restioson** [comment]
I can help with that. If worst comes to worst, I can always send screenshots - I don't really mind either way though, and it should be relatively simple to build the client and send an installer. I will probably send screenshots for the administration panel though, since that would require more complex setup. What operating system and architecture are you on?

**Comment by natastro** [comment]
@Restioson I'm using Windows 10 and have `npm` installed. Let me know if you need any more info, thank you! 😄

**Comment by Gegy** [comment]
@stroudn1 I've compiled an msi for installing on Windows if that works for you :) 
https://drive.google.com/file/d/1kr1o_9pLPuQzwdsg4RC1u_mGpTgWBqTD/view?usp=sharing

Thanks!

**Comment by Restioson** [comment]
Here is a photo of the administration panel:
![image](https://user-images.githubusercontent.com/6688948/86588538-31336580-bf8c-11ea-96f7-a48a39a98fac.png)

I changed the checkboxes to have higher contrast in high contrast stylesheet (toggleable in settings):

![image](https://user-images.githubusercontent.com/6688948/86588610-50ca8e00-bf8c-11ea-8943-8a2afb7f3b90.png)

I've sent these as screenshots because it would be too much work to set it all up on your end, so this is easier for this part. The other stuff should be fine however. You will need to register with a username and password, and for instance ip input `vertex.cf`.

**Comment by natastro** [comment]
## Login screen
![image](https://user-images.githubusercontent.com/17433156/86992614-de1a0800-c16f-11ea-90a8-a6f4609f127e.png)
Light gray login box on dark gray background - needs to be 3:1
- Light gray box: #343434, Background: #2C2C2C, The contrast ratio is: 1.1:1
	- change login box color to: #898989 (or change colors to meet at least 3:1)

Gray fields on gray register box
- Input field box: #3D3D3D, Register box: #343434, The contrast ratio is: 1.1:1
	- Change login box to #898989 (or change colors to meet at least 3:1)

Input field place holders (i.e. Server Instance IP, Choose Username) on gray input fields
- Placeholders: #808080, input fields: #3D3D3D, The contrast ratio is: 2.8:1
	- change text color to #a8a8a8 (or change colors to meet at least 4.5:1)

Error text on gray background [not pictured]
- Text: #D15C4F, Background: #343434, The contrast ratio is: 3.2:1
	- change text color to #EF7A6D (or change colors to meet at least 4.5:1)

Register text on gray background
- Text: #45A0D9, Background: #343434, The contrast ratio is: 4.3:1
	- change text color to #4DA8E1 (or change colors to meet at least 4.5:1)


## Register screen
![image](https://user-images.githubusercontent.com/17433156/86991936-4ec02500-c16e-11ea-835c-4b7eb2282ea3.png)

Light gray register box on dark gray background - needs to be 3:1
- Light gray box: #343434, Background: #2C2C2C, The contrast ratio is: 1.1:1
 	- change register box color to: #898989 (or change colors to meet at least 3:1)

Gray fields on gray register box
- Input field box: #3D3D3D, Register box: #343434, The contrast ratio is: 1.1:1
	- Change register box to #898989 (or change colors to meet at least 3:1)

Input field place holders (i.e. Server Instance IP, Choose Username) on gray input fields
- Placeholders: #808080, input fields: #3D3D3D, The contrast ratio is: 2.8:1
	- change text color to #a8a8a8 (or change colors to meet at least 4.5:1)

Error text on gray background
- Text: #D15C4F, Background: #343434, The contrast ratio is: 3.2:1
	- change text color to #EF7A6D (or change colors to meet at least 4.5:1)

Login text on gray background
- Text: #45A0D9, Background: #343434, The contrast ratio is: 4.3:1
	- change text color to #4DA8E1 (or change colors to meet at least 4.5:1)

@gegy1000 I installed and ran Vertex. I tried registering for an account but received an error (wasn't sure what needed to go in Server Instance IP). Is there a test account I can use? Please let me know if you have any questions on the info above!

**Comment by natastro** [comment]
![image](https://user-images.githubusercontent.com/17433156/86994613-d5780080-c174-11ea-9f05-c0bea17c4c11.png)
Light gray box on dark gray background - needs to be 3:1
- Light gray box: #343434, Background: #2C2C2C, The contrast ratio is: 1.1:1
	- change box color to: #898989 (or change colors to meet at least 3:1)

Dark checkbox on gray background
- Checkbox: #2A2A2A, Background: #343434, The contrast ratio is: 1.2:1
	- change checkbox color to: #7D7D7D (or change colors to meet at least 3:1. FYI - checkmarks need to reach contrast with boxes if color changes)

Placeholder text on gray box
- Text: #7C7C7C, Background: #3D3D3D, The contrast ratio is: 2.6:1
	- change text color to: #A9A9A9 (or change colors to meet at least 4.5:1)

Table outline on dark background
- Outline: #444444, Background: #303030, The contrast ratio is: 1.4:1
	- change outline color to: #797979 (or change color to meet at least 3:1)

Dark gray background (under table) on gray background
- Under table: #303030, Background: #343434, The contrast ratio is: 1.1:1
	- change box color to #7C7C7C (or change colors to meet at least 3:1)

**Comment by Restioson** [comment]
Cool, thanks so much!! I will change the colours and see how it looks. If it turns out to look strange, do you think it'd be acceptable to put it on the high-contrast colours stylesheet and recommend that those who need it turn it on?

**Comment by natastro** [comment]
@Restioson I think best practice is to try and meet contrast first with your colors but you could do that if changing the colors doesn't work.

If it's okay, one tool I like to use is [colorsafe](http://colorsafe.co/). You put in your background color and any text you might have (plus font family - best guess; and font size) and it will give you a palette of colors (gray included) that meet contrast criteria. I also use Paciello Group's [Color Contrast Analyzer](https://developer.paciellogroup.com/resources/contrastanalyser/) which is wonderful too.

One other note is that the [WCAG ](https://www.w3.org/WAI/WCAG21/quickref/)contains two success criteria that differ for contrast. So text & background have to meet 4.5:1 and non-text contrast has to be 3:1. ([SC 1.4.3](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html), [SC 1.4.11](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html))

If you have any questions, please let me know!

**Comment by Restioson** [comment]
Thanks for the feedback. I will try out coloursafe and see how it can be recoloured when I next have time
